api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -655,17 +655,31 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     dbWorkspace.setCdrVersion(fromWorkspace.getCdrVersion());
     dbWorkspace.setDataAccessLevel(fromWorkspace.getDataAccessLevel());
 
-    WorkspaceUserRole permissions = new WorkspaceUserRole();
-    permissions.setRoleEnum(WorkspaceAccessLevel.OWNER);
-    permissions.setWorkspace(dbWorkspace);
-    permissions.setUser(user);
-
-    dbWorkspace.addWorkspaceUserRole(permissions);
-
-    dbWorkspace = workspaceService.saveAndCloneCohortsAndConceptSets(fromWorkspace, dbWorkspace);
-    CloneWorkspaceResponse resp = new CloneWorkspaceResponse();
-    resp.setWorkspace(TO_SINGLE_CLIENT_WORKSPACE_FROM_FC_AND_DB.apply(dbWorkspace, toFcWorkspace));
-    return ResponseEntity.ok(resp);
+    WorkspaceUserRole ownerRole = new WorkspaceUserRole();
+    ownerRole.setRoleEnum(WorkspaceAccessLevel.OWNER);
+    ownerRole.setWorkspace(dbWorkspace);
+    ownerRole.setUser(user);
+
+    dbWorkspace.addWorkspaceUserRole(ownerRole);
+    org.pmiops.workbench.db.model.Workspace savedWorkspace =
+        workspaceService.saveAndCloneCohortsAndConceptSets(fromWorkspace, dbWorkspace);
+
+    if (Boolean.TRUE.equals(body.getIncludeUserRoles())) {
```

Comment:
Can this just be: `if (body.getIncludeUserRoles()) {`

Reply:
We need to tolerate null. Other options include:

- body.getIncludeUserRoles() != null && body.getIncludeUserRoles()
- Optional.ofNullabel(body.getIncludeUserRoles()).orElse(false)

Reply:
Lets do the `Optional.ofNullable()` then? Just to make it more clear why we're doing it this way?

Reply:
Done